binder/src/test/java/io/grpc/binder/SecurityPoliciesTest.java

```diff
@@ -382,14 +383,64 @@ public void testAllOf_succeedsIfAllSecurityPoliciesAllowed() throws Exception {
 
   @Test
   public void testAllOf_failsIfOneSecurityPoliciesNotAllowed() throws Exception {
+    final AtomicBoolean policyCalled = new AtomicBoolean(false);
     policy =
         SecurityPolicies.allOf(
             SecurityPolicies.internalOnly(),
-            SecurityPolicies.permissionDenied("Not allowed SecurityPolicy"));
+            SecurityPolicies.permissionDenied("Not allowed SecurityPolicy"),
+            new SecurityPolicy() {
+              @Override
+              public Status checkAuthorization(int uid) {
+                policyCalled.set(true);
+                throw new AssertionError();
+              }
+            });
 
     assertThat(policy.checkAuthorization(MY_UID).getCode())
         .isEqualTo(Status.PERMISSION_DENIED.getCode());
     assertThat(policy.checkAuthorization(MY_UID).getDescription())
         .contains("Not allowed SecurityPolicy");
+    assertThat(policyCalled.get()).isFalse();
+  }
+
+  @Test
+  public void testAnyOf_succeedsIfAnySecurityPoliciesAllowed() throws Exception {
+    final AtomicBoolean policyCalled = new AtomicBoolean(false);
+    policy =
+        SecurityPolicies.anyOf(
+            SecurityPolicies.internalOnly(),
+            new SecurityPolicy() {
```

Comment:
I don't feel strongly but if we have two copies of this fake perhaps making it a class named RecordingSecurityPolicy or similar would be more self-documenting.

Reply:
Added a helper class to the test.